lms/js_config_types.py

```diff
@@ -17,38 +23,32 @@ class APICourse(TypedDict):
     title: str
 
 
-class APICourses(TypedDict):
-    courses: list[APICourse]
-
-
-class APIStudentStats(TypedDict):
-    display_name: str
-    annotations: int
-    replies: int
-    last_activity: str | None
-
+class APIStudent(TypedDict):
+    id: int
+    display_name: str | None
 
-class APIStudents(TypedDict):
-    students: list[APIStudentStats]
+    annotation_metrics: NotRequired[AnnotationMetrics]
```

Comment:
Now including this type like we do for assignments.

Reply:
Should it be mandatory in this case?

```suggestion
    annotation_metrics: AnnotationMetrics
```

Reply:
Not sure I don't think it should be mandatory, you **could** list students without the metrics but the fact is that it is mandatory now as we only use the type in one place.

:shrug:  I don't have a strong opinion in either direction.

Reply:
Well, if we follow the approach in other types, it should indeed be defined as optional.

What we'll eventually need then, are some API docs that clearly define what every endpoint returns. If the types are internally reused for an endpoint that returns stats and an endpoint that doesn't, this may not be entirely clear otherwwise.